File: SOJO Calculator/VB.NET/Calculator.vb
Public Class Calculator

    Public Const DOT_SIGN As String = ","

    ' ========= OPERATORS =========
    Public Enum Operators
        ADD
        SUBTRACT
        MULTIPLY
        DIVIDE
        MODULO
        PERCENT
        SQUARE
        CUBE
        PI
        POWER
        SQUARE_ROOT
        CUBE_ROOT
        RECIPROCAL
        RND

        NONE
    End Enum

    ' ========= GET OPERATOR SIGN =========
    Public Shared Function getOperatorSign(ByVal op As Operators)
        Dim sign As String = ""

        Select Case op
            Case Operators.ADD
                sign = "+"
            Case Operators.SUBTRACT
                sign = "−"
            Case Operators.MULTIPLY
                sign = "x"
            Case Operators.DIVIDE
                sign = "÷"
            Case Operators.MODULO
                sign = "≡"
            Case Operators.PERCENT
                sign = "%"
            Case Operators.SQUARE
                sign = "x²"
            Case Operators.CUBE
                sign = "x³"
            Case Operators.PI
                sign = "π"
            Case Operators.POWER
                sign = "^"
            Case Operators.SQUARE_ROOT
                sign = "√"
            Case Operators.CUBE_ROOT
                sign = "∛"
            Case Operators.RECIPROCAL
                sign = "1/x"
            Case Operators.RND
                sign = "rnd"
        End Select

        Return sign
    End Function


    ' ========= GET OPERATOR =========
    Public Shared Function getOperator(ByVal key As String)
        Dim op As Calculator.Operators

        Select Case key
            Case "+"
                op = Calculator.Operators.ADD
            Case "−", "-"
                op = Calculator.Operators.SUBTRACT
            Case "x", "×"
                op = Calculator.Operators.MULTIPLY
            Case "÷", "/"
                op = Calculator.Operators.DIVIDE
            Case "mod", "≡" ', "%"
                op = Calculator.Operators.MODULO
            Case "%"
                op = Calculator.Operators.PERCENT
            Case "x²"
                op = Calculator.Operators.SQUARE
            Case "x³"
                op = Calculator.Operators.CUBE
            Case "π"
                op = Calculator.Operators.PI
            Case "^"
                op = Calculator.Operators.POWER
            Case "√"
                op = Calculator.Operators.SQUARE_ROOT
            Case "∛"
                op = Calculator.Operators.CUBE_ROOT
            Case "1/x"
                op = Calculator.Operators.RECIPROCAL
            Case "rnd"
                op = Calculator.Operators.RND

            Case Else
                op = Calculator.Operators.NONE
        End Select

        Return op
    End Function

    ' ========= CALCULATION OF CUBED ROOT =========
    Public Shared Function CubedRoot(ByVal dNum As Double) As Double
        Return CType(CType(dNum ^ (1 / 3), Decimal), Double)
    End Function


    ' ========= CALCULATE RESULT =========
    Public Shared Function Calculate(ByVal firstValue As Double, ByVal secondValue As Double, ByVal operation As Calculator.Operators) As Double
        Dim result As Double = 0

        Select Case operation
            Case Calculator.Operators.ADD
                result = firstValue + secondValue
            Case Calculator.Operators.SUBTRACT
                result = firstValue - secondValue
            Case Calculator.Operators.MULTIPLY
                result = firstValue * secondValue
            Case Calculator.Operators.DIVIDE
                result = firstValue / secondValue
            Case Calculator.Operators.MODULO
                result = firstValue Mod secondValue
            Case Calculator.Operators.PERCENT
                result = firstValue / 100
            Case Calculator.Operators.SQUARE
                result = firstValue * firstValue
            Case Calculator.Operators.CUBE
                result = firstValue * firstValue * firstValue
            Case Calculator.Operators.PI
                result = Math.PI
            Case Calculator.Operators.POWER
                result = Math.Pow(firstValue, secondValue)
            Case Operators.SQUARE_ROOT
                result = Math.Sqrt(firstValue)
            Case Operators.CUBE_ROOT
                result = Calculator.CubedRoot(firstValue)
            Case Operators.RECIPROCAL
                result = 1 / firstValue

            Case Else
                Debug.Print("Case Else")
        End Select

        Return result
    End Function


    ' ========= FORMAT VALUE =========
    Public Shared Function FormatValue(ByVal input As String, Optional ByVal forceFormat As Boolean = False) As String
        If input.Length > 1 And input.Substring(0, 1) = "0" And input.Contains(DOT_SIGN) = False Then
            Return input.Substring(1)
        ElseIf input.Contains(DOT_SIGN) Then
            If forceFormat Then
                Return Format(CDbl(input), "##,##0.#####################")
            Else
                Return input
            End If
        Else
            Return Format(CDbl(input), "##,##0.#####################")
        End If
    End Function

End Class

File: SOJO Calculator/C#.NET/Calculator.cs
using System;
using System.Diagnostics;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace SOJO_Calculator
{
    public class Calculator
    {

        public const string DOT_SIGN = ",";
        public const float BASE_FONT_SIZE = 24f;

        // ========= OPERATORS =========
        public enum Operators
        {
            ADD,
            SUBTRACT,
            MULTIPLY,
            DIVIDE,
            MODULO,
            PERCENT,
            SQUARE,
            CUBE,
            PI,
            POWER,
            SQUARE_ROOT,
            CUBE_ROOT,
            RECIPROCAL,
            RND,
            EQUAL,

            NONE
        }

        // ========= GET OPERATOR SIGN =========
        public static object getOperatorSign(Operators op)
        {
            string sign = "";

            switch (op)
            {
                case Operators.ADD:
                    {
                        sign = "+";
                        break;
                    }
                case Operators.SUBTRACT:
                    {
                        sign = "−";
                        break;
                    }
                case Operators.MULTIPLY:
                    {
                        sign = "x";
                        break;
                    }
                case Operators.DIVIDE:
                    {
                        sign = "÷";
                        break;
                    }
                case Operators.MODULO:
                    {
                        sign = "≡";
                        break;
                    }
                case Operators.PERCENT:
                    {
                        sign = "%";
                        break;
                    }
                case Operators.SQUARE:
                    {
                        sign = "x²";
                        break;
                    }
                case Operators.CUBE:
                    {
                        sign = "x³";
                        break;
                    }
                case Operators.PI:
                    {
                        sign = "π";
                        break;
                    }
                case Operators.POWER:
                    {
                        sign = "^";
                        break;
                    }
                case Operators.SQUARE_ROOT:
                    {
                        sign = "√";
                        break;
                    }
                case Operators.CUBE_ROOT:
                    {
                        sign = "∛";
                        break;
                    }
                case Operators.RECIPROCAL:
                    {
                        sign = "1/x";
                        break;
                    }
                case Operators.RND:
                    {
                        sign = "rnd";
                        break;
                    }
                case Operators.EQUAL:
                    {
                        sign = "=";
                        break;
                    }
            }

            return sign;
        }


        // ========= GET OPERATOR =========
        public static object getOperator(string key)
        {
            Operators op;

            switch (key ?? "")
            {
                case "+":
                    {
                        op = Operators.ADD;
                        break;
                    }
                case "−":
                case "-":
                    {
                        op = Operators.SUBTRACT;
                        break;
                    }
                case "x":
                case "×":
                    {
                        op = Operators.MULTIPLY;
                        break;
                    }
                case "÷":
                case "/":
                    {
                        op = Operators.DIVIDE;
                        break;
                    }
                case "mod":
                case "≡": // , "%"
                    {
                        op = Operators.MODULO;
                        break;
                    }
                case "%":
                    {
                        op = Operators.PERCENT;
                        break;
                    }
                case "x²":
                    {
                        op = Operators.SQUARE;
                        break;
                    }
                case "x³":
                    {
                        op = Operators.CUBE;
                        break;
                    }
                case "π":
                    {
                        op = Operators.PI;
                        break;
                    }
                case "^":
                    {
                        op = Operators.POWER;
                        break;
                    }
                case "√":
                    {
                        op = Operators.SQUARE_ROOT;
                        break;
                    }
                case "∛":
                    {
                        op = Operators.CUBE_ROOT;
                        break;
                    }
                case "1/x":
                    {
                        op = Operators.RECIPROCAL;
                        break;
                    }
                case "rnd":
                    {
                        op = Operators.RND;
                        break;
                    }
                case "=":
                    {
                        op = Operators.EQUAL;
                        break;
                    }

                default:
                    {
                        op = Operators.NONE;
                        break;
                    }
            }

            return op;
        }

        // ========= CALCULATION OF CUBED ROOT =========
        public static double CubedRoot(double dNum)
        {
            return (double)(decimal)Math.Pow(dNum, 1d / 3d);
        }


        // ========= CALCULATE RESULT =========
        public static double Calculate(double firstValue, double secondValue, Operators operation)
        {
            double result = 0d;

            switch (operation)
            {
                case Operators.ADD:
                    {
                        result = firstValue + secondValue;
                        break;
                    }
                case Operators.SUBTRACT:
                    {
                        result = firstValue - secondValue;
                        break;
                    }
                case Operators.MULTIPLY:
                    {
                        result = firstValue * secondValue;
                        break;
                    }
                case Operators.DIVIDE:
                    {
                        result = firstValue / secondValue;
                        break;
                    }
                case Operators.MODULO:
                    {
                        result = firstValue % secondValue;
                        break;
                    }
                case Operators.PERCENT:
                    {
                        result = firstValue / 100d;
                        break;
                    }
                case Operators.SQUARE:
                    {
                        result = firstValue * firstValue;
                        break;
                    }
                case Operators.CUBE:
                    {
                        result = firstValue * firstValue * firstValue;
                        break;
                    }
                case Operators.PI:
                    {
                        result = Math.PI;
                        break;
                    }
                case Operators.POWER:
                    {
                        result = Math.Pow(firstValue, secondValue);
                        break;
                    }
                case Operators.SQUARE_ROOT:
                    {
                        result = Math.Sqrt(firstValue);
                        break;
                    }
                case Operators.CUBE_ROOT:
                    {
                        result = CubedRoot(firstValue);
                        break;
                    }
                case Operators.RECIPROCAL:
                    {
                        result = 1d / firstValue;
                        break;
                    }

                default:
                    {
                        Debug.Print("Case Else");
                        break;
                    }
            }

            return result;
        }


        // ========= FORMAT VALUE =========
        public static string FormatValue(string input, bool forceFormat = false)
        {
            if (input.Length > 1 & input.Substring(0, 1) == "0" & input.Contains(DOT_SIGN) == false)
            {
                return input.Substring(1);
            }
            else if (input.Contains(DOT_SIGN))
            {
                if (forceFormat)
                {
                    return Strings.Format(Conversions.ToDouble(input), "##,##0.#####################");
                }
                else
                {
                    return input;
                }
            }
            else
            {
                return Strings.Format(Conversions.ToDouble(input), "##,##0.#####################");
            }
        }

    }
}